DataStructures/List.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+
+namespace DataStructures
+{
+    public class List<T>
+    {
+        private T[] items;
+        private int count;
+
+        public List()
+        {
+            items = new T[5];
+            count = 0;
+        }
+        public List(int index)
+        {
+            items = new T[index];
+            count = index;
+        }
+
+        public void Add(T item)
+        {
+            items[count] = item;
+            count++;
+            if (count == items.Length)
```

Comment:
This logic should be first in the method, for example if I create a list of 15 spaces `new List<int>(15)` and then I insert elements until fill those 15 spaces, you are incrementing the array immediately after I insert the 15 element, but that is wrong in my particular case: I explicitly said that I want 15 spaces `new List<int>(15)` because I will make sure that my code does not insert more than 15 elements, and if I will not insert more than 15 elements why should I have a List that internally have an array with more than 15 elements.


If this logic is at the start, you will make sure to increment the list only if the user decide to put a 16 element.

Reply:
You say that the logic of line 25 should be begin before line 23...
Well, now that I see it, I think the same, I'll fix it.